### PR TITLE
Add guidelines for contributors.

### PR DESCRIPTION
**Pull request by syves**
Please add comments about what your conventions are.

Do you like commits to be squished?

### CODE REVIEW

**Comment by syves** [comment]
I can not rerun this build without write permissions.
"gpg: no valid OpenPGP data found."

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Contributing
+
+1.  Update local master branch:
```
Comment (allquantor) [inline CONTRIBUTING.md]: Contributions should start from `develop` branch.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+        $ git checkout master
```
Comment (allquantor) [inline CONTRIBUTING.md]: Would be nice to indicate that this should be a fork and that an upstream have to be configured. A link would be enough.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+        $ git pull upstream master
+
+2.  Create feature branch:
+
+        $ git checkout -b feature-x
+
+3.  Make one or more atomic commits, and ensure that each commit has a
+    descriptive commit message. Commit messages should be line wrapped
+    at 72 characters.
```
Comment (allquantor) [inline CONTRIBUTING.md]: Why is this necessary?

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+4.  Run tests, and address any errors. Preferably, fix commits in place
```
Comment (allquantor) [inline CONTRIBUTING.md]: Please more details on how and which tests to run.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    using `git rebase` or `git commit --amend` to make the changes easier to
```
Comment (allquantor) [inline CONTRIBUTING.md]: This is not necessary to be in here.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    review.
+
+5.  Push:
+
+        $ git push origin feature-x
+
+6.  Open a pull request. Pull requests should be small enough to review in one day.
```
Comment (allquantor) [inline CONTRIBUTING.md]: Not necessary.

File: CONTRIBUTING.md (hunk 1 above)
Comment (syves) [reply CONTRIBUTING.md]: 🆗

File: CONTRIBUTING.md (hunk 2 above)
Comment (syves) [reply CONTRIBUTING.md]: 🆗

File: CONTRIBUTING.md (hunk 3 above)
Comment (syves) [reply CONTRIBUTING.md]: I'll remove it.

File: CONTRIBUTING.md (hunk 5 above)
Comment (syves) [reply CONTRIBUTING.md]: 🆗

File: CONTRIBUTING.md (hunk 6 above)
Comment (syves) [reply CONTRIBUTING.md]: 🆗

**Comment by syves** [comment]
Changes have been applied.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+        $ git checkout develop
+        $ git checkout -b feature-x
+
+4.  Make one or more atomic commits, and ensure that each commit has a
+    descriptive commit message.
+
+5. Update existing Spec or create one.    
+
+6.  Run tests, and address any errors.
```
Comment (allquantor) [inline CONTRIBUTING.md]: The testing concept for the client is a little bit tricky. Since we need to test it on a network, we run a temp arweave network wich an modified mining speed to make integration tests possible.
You can get familiar with the step by consulting the CI config here: https://github.com/toknapp/arweave4s/blob/master/.circleci/config.yml
Feel free to describe the process here - since just run integration test would fail without running a network.

Comment (syves) [reply CONTRIBUTING.md]: Thank you. I was planning to explore the yml today.
